### ISSUE TITLE
Test failure: JIT\\Methodical\\structs\\systemvbringup\\structinregs\\structinregs.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#36967]
failed in job: [runtime-coreclr jitstress2-jitstressregs 20200524.1](https://dev.azure.com/dnceng/public/_build/results?buildId=658473&view=ms.vss-test-web.build-test-results-tab&runId=20435328&resultId=106408&paneView=debug)

Error message
~~~
Return code: 1
Raw output file: C:\h\w\A6980933\w\B3DF09DB\e\JIT\Methodical\Reports\JIT.Methodical\structs\systemvbringup\structinregs\structinregs.output.txt
Raw output:
BEGIN EXECUTION
 "C:\h\w\A6980933\p\corerun.exe" structinregs.dll 
From Program0:test1!
test1 Result: 3
From Program0:test2!
test2 Result: 3
From Program0:test3!
test3 Result: 7
From Program0:test4!
test4 Result: 11
From Test5 members: 10 11 12 33
From Test5: Res 33
test5 Result: 42
From Test6 members: 21 22
From Test6: Res 43
test6 Result: 57
From Test7 members: 21 22
From Test7: Res 43
test7Result: 57
Foo2:Foo2:Foo2!!!
t13Res: 333
test13 Result: 333
t14 Res: 444
test14 Result: 444
t10 Res: True, 7
test10 Result: 15
t11 Res: 78, 87
test11 Result: 165
t12Res: 101
test12 Result: 101
test1: 36
test1 Result: 36
test2: 19
test2 Result: 19
test3: structinreg.Foo2 structinreg.Foo2 structinreg.Foo2 structinreg.Foo2
test3 Result: 10
test4 Res: 300
test4 Result: 300
test5 Res: 10
test5 Result: 10
t7 Res: 130
test7 Result: 130
t15 Res: 136
test15 Result: 136
test1Res: 36
test1 Result: 36
test2Res: 19
test2 Result: 19
test3Res: System.Object System.Object System.Object System.Object
test3 Result: 164711009
test4Res: 3
test4 Result: 300
test5Res: 36
S1: 1, 2, 3, 4
S2: 1, 2, 3
S3: 1, 2, 3
S4: 1, 2
S5: 1, 2
S6: 1, 2, 3, 4
S7: 1, 2, 3
S8: 1, 2
S9: 1, 2, 3, 4
S10: 1, 2, 3, 4, 5, 6, 7, 8
S11: 1, 2, 3, 4, 5
S12: 1, 2, 3, 4, 5, 6, 7, 8, 9
S13: 1, 2
S14: 1, 2
S15: 1, 2, 3, 4, 5, 6, 7, 8, 9
S16: 1, 2
S17: 1, 2
S18: 1, 2, 3
S19: 1, 2, 3, 4
S20: 1, 2, 3, 4
S30: 1, 2, 3, 4, 5, 6
S1: 1, 2, 3, 4
S1R: 1, 2, 3, 4
S2: 1, 2, 3
S2R: 1, 2, 3
S3: 1, 2, 3
S3R: 1, 2, 3
S4: 1, 2
S4R: 1, 1.4779274E+28
Exception of type 'System.Exception' was thrown.
Native S1: 1, 2, 3, 4
Native S2: 1, 2, 3.000000
Native S3: 1, 2, 3.000000
Native S4: 1, 2.000000
Native S5: 1, 2.000000
Native S6: 1, 2, 3, 4
Native S7: 1.000000, 2, 3
Native S8: 1.000000, 2
Native S9: 1, 2, 3.000000, 4.000000
Native S10: 1, 2, 3, 4, 5, 6, 7, 8
Native S11: 1, 2, 3, 4, 5.000000
Native S12: 1, 2, 3, 4, 5, 6, 7, 8, 9
Native S13: 1, 2
Native S13: 1, 2
Native S15: 1, 2, 3, 4, 5, 6, 7, 8, 9
Native S16: 1, 2
Native S17: 1.000000, 2.000000
Native S18: 1.000000, 2, 3.000000
Native S19: 1, 2.000000, 3, 4.000000
Native S20: 1, 2, 3, 4
Native S30: 1, 2, 3, 4, 5, 6
Native S1: 1, 2, 3, 4
Native S2: 1, 2, 3.000000
Native S3: 1, 2, 3.000000
Native S4: 1, 2.000000
Expected: 100
Actual: -1
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\A6980933\p
> C:\h\w\A6980933\w\B3DF09DB\e\JIT\Methodical\structs\systemvbringup\structinregs\structinregs.cmd
Expected: True
Actual: False


Stack trace
   at JIT_Methodical._structs_systemvbringup_structinregs_structinregs_._structs_systemvbringup_structinregs_structinregs_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x86.Checked\TestWrappers\JIT.Methodical\JIT.Methodical.XUnitWrapper.cs:line 175031
~~~

**Comment by BruceForstall** [comment]
Failure mode: `CoreCLR Windows_NT x86 Checked jitstress2_jitstressregs4 @ Windows.10.Amd64.Open`

@dotnet/jit-contrib

**Comment by CarolEidt** [comment]
Fixed by #37062
